arm/Microsoft.RecoveryServices/vaults/deploy.bicep

```diff
@@ -131,33 +124,47 @@ resource rsv 'Microsoft.RecoveryServices/vaults@2021-08-01' = {
     tier: 'Standard'
   }
   properties: {}
+}
 
-  resource rsv_vaultstorageconfig 'backupstorageconfig@2020-02-02' = {
-    name: 'vaultstorageconfig'
-    properties: {
-      StorageModelType: vaultStorageType
-      CrossRegionRestoreFlag: enableCRR
-    }
+module rsv_backupStorageConfig 'backupStorageConfig/deploy.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-BackupStorageConfig'
+  params: {
+    recoveryVaultName: rsv.name
+    vaultStorageType: backupStorageConfig.vaultStorageType
+    enableCRR: backupStorageConfig.enableCRR
   }
-
-  resource rsv_protectionContainers 'protectionContainers@2016-12-01' = [for (protectionContainer, index) in protectionContainers: {
-    name: protectionContainer.Name
-    location: resourceGroup().location
-    properties: {
-      sourceResourceId: (empty(protectionContainer.sourceResourceId) ? json('null') : protectionContainer.sourceResourceId)
-      friendlyName: (empty(protectionContainer.friendlyName) ? json('null') : protectionContainer.friendlyName)
-      backupManagementType: (empty(protectionContainer.backupManagementType) ? json('null') : protectionContainer.backupManagementType)
-      containerType: (empty(protectionContainer.containerType) ? json('null') : protectionContainer.containerType)
-    }
-  }]
-
-  resource rsv_backupPolicies 'backupPolicies@2019-06-15' = [for (protectionPolicy, index) in backupPolicies: {
-    name: protectionPolicy.name
-    location: resourceGroup().location
-    properties: protectionPolicy.Properties
-  }]
+  dependsOn: [
+    rsv
+  ]
 }
 
+module rsv_protectionContainers 'protectionContainers/deploy.bicep' = [for (protectionContainer, index) in protectionContainers: {
```

Comment:
We have a missing level for `backupFabrics `here that we should add since `protectionContainers `is in there. 
We should then have a _backupFabrics_ folder under _vaults_ with corresponding bicep template and readme. Then have a _protectionContainers_ subfolder in there hosting corresponding bicep template and readme as well. 

You can refer to the `Microsoft.Storage\storageAccounts` module as this is similar to what done for blob, file, queue and table services there.

Reply:
Discussed over IM that backupFabrics cannot be deployed by itself as a child resource